src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -305,6 +305,7 @@ void ShenandoahConcurrentGC::entry_init_mark() {
                               "init marking");
 
   op_init_mark();
+  ShenandoahHeap::heap()->propagate_gc_state_to_all_threads();
```

Comment:
I would say move it downwards into `op_init_mark` and wrap with its own subtimer, e.g.  with `ShenandoahGCPhase phase(ShenandoahPhaseTimings::init_propagate_gcstate);`.